driver/src/test/java/org/neo4j/driver/internal/util/MetadataExtractorTest.java

```diff
@@ -303,20 +313,87 @@ void shouldBuildResultSummaryWithNotifications() {
         assertEquals("Neo.GoodNotification", secondNotification.code());
         assertEquals("Good", secondNotification.title());
         assertEquals("INFO", secondNotification.severity());
-        assertEquals(new InternalInputPosition(1, 2, 3), secondNotification.position());
+        assertTrue(secondNotification.inputPosition().isEmpty());
+        assertNull(secondNotification.position());
+
+        assertEquals(2, summary.gqlStatusObjects().size());
+        var gqlStatusObjectsIterator = summary.gqlStatusObjects().iterator();
+        var firstGqlStatusObject = (Notification) gqlStatusObjectsIterator.next();
+        var secondGqlStatusObject = (Notification) gqlStatusObjectsIterator.next();
+        assertEquals(firstNotification, firstGqlStatusObject);
+        assertEquals(secondNotification, secondGqlStatusObject);
+    }
+
+    @Test
+    @SuppressWarnings({"deprecation", "OptionalGetWithoutIsPresent"})
+    void shouldBuildResultSummaryWithGqlStatusObjects() {
```

Comment:
It might be good to also exercises filling the default for  missing values like the diagnostic_record, diagnostic_record.OPERATION, and so on.

Reply:
Good point, this have been added. 👍